mergebot_poller.py

```diff
@@ -1,47 +1,131 @@
-"""TODO(jasonkuster): DO NOT SUBMIT without one-line documentation for mergebot_poller.
-
-TODO(jasonkuster): DO NOT SUBMIT without a detailed description of mergebot_poller.
+"""
+mergebot_poller is a utility which polls an SCM and follows commands.
 """
 
-import mergebot
+from github_helper import GithubHelper
+import merge
 import Queue
+from threading import Thread
+import time
+
+# TODO(jasonkuster): Fetch authorized users from somewhere official.
+AUTHORIZED_USERS = ["davorbonaci"]
+BOT_NAME = 'merge-bot'
+ORG = 'apache'
+
+
+class MergebotPoller(object):
+    """
+    MergebotPoller is a class which handles the polling of a configured SCM.
+    """
+
+    SCM_POLLERS = {'github': poll_github}
+    SCM_MERGERS = {'github': merge.merge_git}
+    CMD_MERGE = {'github': merge_git}
+    CMDS = {'merge': CMD_MERGE}
+
+    def __init__(self, config):
+        self.config = config
+        # Instantiate the two variables used for tracking work.
+        # Thread-safe because of the GIL.
+        self.work_queue = Queue.Queue()
+        self.known_work = set()
+        # Set up a github helper for handling network requests, etc.
+        self.github_helper = GithubHelper(ORG, self.config['repository'])
+
+    def poll(self):
+        """
+        Kicks off polling of the SCM. Reads SCM type from config and starts the
+        appropriate poller.
 
-GITHUB_SECRET = '../../github_auth/apache-beam.secret'
+        """
+        print('Starting merge poller for {}.'.format(self.config['name']))
+        self.SCM_POLLERS[self.config['scm_type']]()
 
-class MergebotPoller:
-  scm_type = ""
-  repository = ""
-  branch = ""
-  scm_pollers = { "github": poll_github }
-  scm_mergers = { "github": mergebot.merge_git }
-  work_queue = Queue.Queue()
+    def poll_github(self):
```

Comment:
The use of the map `SCM_POLLERS` and the implementation of methods like `poll` to dispatch to them seems like a smell. I'd suggest using multiple objects, so that the built-in method dispatch takes care of this.

```
class MergebotPoller(object):
  """Base class for pollers (may be unnecessary if this doesn't have anything interesting)."""

class GithubPoller(MergebotPoller):
  def poll(self):
    ...
```

Then you can have a top-level factory method for creating the right kind of poller:

```
def CreatePoller(config):
  if self.config['scm_type'] == 'github':
    return new GithubPoller(config)
  ...
```